src/mesh/mesh_refinement.cpp

```diff
@@ -38,6 +38,52 @@
 
 namespace parthenon {
 
+// TODO(Yurlunger) Identify an idiomatic way of applying reflective
+// boundary conditions in the boundary condition refactor code
+void applyBounds(std::shared_ptr<MeshBlock> pmb, ParArrayND<Real> &a,
+                 const IndexRange &ib, const IndexRange &jb) {
+  // applyBounds() is a Hack.  This needs to go, see TODO above.
+
+  if (pmb->boundary_flag[BoundaryFace::outer_x1] == BoundaryFlag::reflect) {
+    for (int n = 0; n < a.GetDim(4); n++) {
+      for (int j = 0; j <= jb.e + NGHOST; j++) {
+        for (int i = ib.e + 1; i <= ib.e + NGHOST; i++) {
+          a(n, 0, j, i) = a(n, 0, j, 2 * ib.e - i + 1);
+        }
+      }
+    }
+  }
+
+  if (pmb->boundary_flag[BoundaryFace::inner_x1] == BoundaryFlag::reflect) {
+    for (int n = 0; n < a.GetDim(4); n++) {
+      for (int j = 0; j <= jb.e + NGHOST; j++) {
+        for (int i = 0; i < ib.s; i++) {
+          a(n, 0, j, i) = a(n, 0, j, 2 * ib.s - i - 1);
+        }
+      }
+    }
+  }
+
+  if (pmb->boundary_flag[BoundaryFace::outer_x2] == BoundaryFlag::reflect) {
+    for (int n = 0; n < a.GetDim(4); n++) {
+      for (int j = jb.e + 1; j <= jb.e + NGHOST; j++) {
+        for (int i = 0; i <= ib.e + NGHOST; i++) {
+          a(n, 0, j, i) = a(n, 0, 2 * jb.e - j + 1, i);
+        }
+      }
+    }
+  }
+
+  if (pmb->boundary_flag[BoundaryFace::inner_x2] == BoundaryFlag::reflect) {
+    for (int n = 0; n < a.GetDim(4); n++) {
+      for (int j = 0; j < jb.s; j++) {
+        for (int i = 0; i <= ib.e + NGHOST; i++) {
+          a(n, 0, j, i) = a(n, 0, 2 * jb.s - j - 1, i);
+        }
+      }
+    }
+  }
+}
```

Comment:
This actually needs to be done for reflecting, outflow, and "user," not just reflecting. But I'll deal with it in my MR.

Reply:
Thanks!